OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -297,58 +304,64 @@ bool ShouldExplode(World world)
 				return true;
 
 			// After first bounce, check for targets each tick
-			if (remainingBounces < info.BounceCount && AnyValidTargetsInRadius(world, pos, info.Width, args.SourceActor, true))
+			if (remainingBounces < info.BounceCount && AnyValidTargetsInRadius(world, pos, info.Width, Args.SourceActor, true))
 				return true;
 
 			return false;
 		}
 
-		public IEnumerable<IRenderable> Render(WorldRenderer wr)
+		public virtual IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
 			if (info.ContrailLength > 0)
 				yield return contrail;
 
-			if (anim == null || ticks >= length)
+			if (ticks >= length)
```

Comment:
```suggestion
			if (FlightLengthReached)
```

So the variable gets used